src/t_hash.c

```diff
@@ -922,3 +922,158 @@ void hscanCommand(client *c) {
         checkType(c,o,OBJ_HASH)) return;
     scanGenericCommand(c,o,cursor);
 }
+
+#define HRANDMEMBER_SUB_STRATEGY_MUL 3
+
+void hrandmemberWithCountCommand(client *c, long l) {
+    unsigned long entryCount, hashSize;
+    int uniq = 1;
+    hashTypeIterator *hi;
+    robj *hash;
+    dict *d;
+    double randomDouble;
+    double threshold;
+    unsigned long index = 0;
+
+    if ((hash = lookupKeyReadOrReply(c,c->argv[1],shared.null[c->resp]))
+        == NULL || checkType(c,hash,OBJ_HASH)) return;
+
+    if(l >= 0) {
+        entryCount = (unsigned long) l;
+    } else {
+        entryCount = -l;
+        uniq = 0;
+    }
+
+    hashSize = hashTypeLength(hash);
+    if(entryCount > hashSize)
+        entryCount = hashSize;
+    addReplyMapLen(c, entryCount);
+    hi = hashTypeInitIterator(hash);
+
+    if(hash->encoding == OBJ_ENCODING_ZIPLIST) {
+        while (hashTypeNext(hi) != C_ERR && entryCount != 0) {
+            randomDouble = ((double)rand()) / RAND_MAX;
+            threshold = ((double)entryCount) / (hashSize - index);
+            if(randomDouble < threshold){
+                entryCount--;
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_KEY);
+                addHashIteratorCursorToReply(c, hi, OBJ_HASH_VALUE);
+            }
+        
+            index ++;
```

Comment:
```suggestion
            index++;
```

Should the space be deleted?